prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -409,38 +491,41 @@ def rerank_datasets(self, dataset_list: list[str], prompt_spec: PromptSpec):
                 no suitable dataset is found or if the confidence level
                 in the recommendation is low.
         """
-        dataset_info_dict = self.get_all_dataset_infos(dataset_list)
-
-        if len(dataset_info_dict.keys()) == 0:
-            return None
-        prompt = construct_prompt_for_dataset_reranking(
-            prompt_spec.instruction, prompt_spec.examples, dataset_info_dict
+        dataset_selection_prompt = construct_prompt_for_dataset_reranking(
+            prompt_spec.instruction, prompt_spec.examples, datasets_info_dict
         )
-        response = parse_prompt_to_fields(prompt=prompt, module_name="rerank")
-
-        dataset_name, config_name, confidence_level = (
-            response["dataset_name"],
-            response["config_name"],
-            response["confidence_level"],
+        dataset_name = self.get_rerank_with_highest_votes(
+            prompt=dataset_selection_prompt, infos_dict=datasets_info_dict
         )
-        if (
-            dataset_name not in dataset_info_dict
-            or config_name not in dataset_info_dict[dataset_name]["configs"]
-        ):
-            logger.warning("LLM hallucinated dataset/config name")
-            return None
-        if confidence_level == "low":
-            logger.warning("Confidence in retrieved dataset is low.")
-            return None
 
-        return dataset_info_dict[dataset_name]["configs"][config_name]
+        if dataset_name is None:
+            return None, None
+
+        time.sleep(10)  # To avoid rate limiting
+
+        if len(datasets_info_dict[dataset_name]["configs"].keys()) == 1:
+            config_name = list(datasets_info_dict[dataset_name]["configs"].keys())[0]
+
+        else:
+            curr_dataset = datasets_info_dict[dataset_name]
+            if len(curr_dataset["configs"]) > 10:
+                curr_dataset["configs"] = dict(
+                    random.sample(list(curr_dataset["configs"].items()), 10)
+                )
+            config_selection_prompt = construct_prompt_for_dataset_reranking(
+                prompt_spec.instruction,
+                prompt_spec.examples,
+                curr_dataset,
+                is_config=True,
+            )
+            config_name = self.get_rerank_with_highest_votes(
```

Comment:
if `config_name` came out to be `None`, we still want to return None, None right? also a helpful case for logger warning

Reply:
Yeah if config name came out None it would automatically get handled in the calling function, but I can add a comment to make this clearer